controllers/serveController.js

```diff
@@ -1,150 +1,43 @@
-  logger.debug('...getting asset by claim Id...');
-  return new Promise((resolve, reject) => {
-    // 1. check locally for claim
-    checkForLocalAssetByClaimId(fullClaimId, name)
-    .then(dataValues => {
-      // if a result was found, return early with the result
-      if (dataValues) {
-        logger.debug('found a local file for this name and claimId');
-        resolve(dataValues);
-        return;
-      }
-      logger.debug('no local file found for this name and claimId');
-      // 2. if no local claim, resolve and get the claim
-      db.Claim
-        .resolveClaim(name, fullClaimId)
-        .then(resolveResult => {
-          // if no result, return early (claim doesn't exist or isn't free)
-          if (!resolveResult) {
-            resolve(NO_CLAIM);
-            return;
-          }
-          logger.debug('resolve result >> ', resolveResult.dataValues);
-          let fileRecord = {};
-          // get the claim
-          lbryApi.getClaim(`${name}#${fullClaimId}`)
-          .then(getResult => {
-            logger.debug('getResult >>', getResult);
-            fileRecord = createFileRecord(resolveResult);
-            fileRecord = addGetResultsToFileRecord(fileRecord, getResult);
-            // insert a record in the File table & Update Claim table
-            return db.File.create(fileRecord);
-          })
-          .then(() => {
-            logger.debug('File record successfully updated');
-            resolve(fileRecord);
-          })
-          .catch(error => {
-            reject(error);
-          });
-        })
-        .catch(error => {
-          reject(error);
-        });
-    })
-    .catch(error => {
-      reject(error);
-    });
-  });
-}
-
-function chooseThumbnail (claimInfo, defaultThumbnail) {
-  if (!claimInfo.thumbnail || claimInfo.thumbnail.trim() === '') {
-    return defaultThumbnail;
-  }
-  return claimInfo.thumbnail;
-}
+const NO_FILE = 'NO_FILE';
 
 module.exports = {
-  getAssetByClaim (claimName, claimId) {
-    logger.debug(`getAssetByClaim(${claimName}, ${claimId})`);
+  getClaimId (channelName, channelId, name, claimId) {
+    if (channelName) {
+      return module.exports.getClaimIdByChannel(channelName, channelId, name);
+    } else {
+      return module.exports.getClaimIdByClaim(name, claimId);
+    }
+  },
+  getClaimIdByClaim (claimName, claimId) {
+    logger.debug(`getClaimIdByClaim(${claimName}, ${claimId})`);
     return new Promise((resolve, reject) => {
-      db.Claim.getLongClaimId(claimName, claimId) // 1. get the long claim id
-        .then(result => {  // 2. get the asset using the long claim id
-          logger.debug('long claim id ===', result);
-          if (result === NO_CLAIM) {
-            logger.debug('resolving NO_CLAIM');
-            resolve(NO_CLAIM);
-            return;
-          }
-          resolve(getAssetByLongClaimId(result, claimName));
+      db.Claim.getLongClaimId(claimName, claimId) // get the long claim id
```

Comment:
No need to wrap `resolve` and `reject` if you are just passing them the same parameter. You can just do:

`db.Claim.getLongClaimId(claimName, claimId).then(resolve, reject)`